### PR TITLE
alien vault otx taxii feed

### PR DESCRIPTION
**Pull request by roysagi**
## Status
- [x] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/22115

## Description
alien vault otx taxii feed

### CODE REVIEW

**Comment by roysagi** [comment]
Still missing default params info but the rest is done

**Comment by roysagi** [comment]
> Still missing default params info but the rest is done

Done

File: Packs/FeedAlienVault/Integrations/FeedAlienVaultOTXTaxii/FeedAlienVaultOTXTaxii.yml [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  required: false
+  type: 8
+- display: Trust any certificate (not secure)
+  name: insecure
+  required: false
+  type: 8
+- display: Use system proxy settings
+  name: proxy
+  required: false
+  type: 8
```
Comment (ronykoz) [inline Packs/FeedAlienVault/Integrations/FeedAlienVaultOTXTaxii/FeedAlienVaultOTXTaxii.yml]: those two should be at the bottom

**Comment by roysagi** [comment]
> Add UT, don't we want to add default mapping here?

I don't think I have anything to test in the unit tests here... except maybe indicator parsing. The decoder functionality is basically tested in the playbook (by actually creating indicators).

As for mapping I'll sit with @David-BMS about what else cones out of the decoders.

File: Packs/FeedAlienVault/Integrations/FeedAlienVaultOTXTaxii/FeedAlienVaultOTXTaxii.yml (hunk 1 above)
Comment (roysagi) [reply Packs/FeedAlienVault/Integrations/FeedAlienVaultOTXTaxii/FeedAlienVaultOTXTaxii.yml]: done

File: Packs/FeedAlienVault/Integrations/FeedAlienVaultOTXTaxii/FeedAlienVaultOTXTaxii.py [hunk 2]
```diff
@@ -0,0 +1,667 @@
...
+                raise Exception(e)
+
+            else:
+                continue
+
+        index = 0
+        only_indicator_list = []  # type:List
+        for raw_response in taxii_iter:
+            _, res = client.decode_indicators(raw_response.content)
+            parsed_list, only_indicator_list = parse_indicators(res, only_indicator_list)
```
Comment (ronykoz) [inline Packs/FeedAlienVault/Integrations/FeedAlienVaultOTXTaxii/FeedAlienVaultOTXTaxii.py]: you are not using only_indicator_list anywhere...
And why are you sending an empty list to the parse_indicators function?
I think that this line should be 
```suggestion
            parsed_list, _ = parse_indicators(res, indicator_list)
```
Please check that

File: Packs/FeedAlienVault/Integrations/FeedAlienVaultOTXTaxii/FeedAlienVaultOTXTaxii.py [hunk 3]
```diff
@@ -0,0 +1,667 @@
...
+
+    Returns:
+        list,list. A list of parsed indicators and an updated list of all indicators polled
+    """
+    parsed_indicator_list = []  # type: List
+    for indicator in sub_indicator_list:
+        if indicator['indicator'] in full_indicator_list:
+            continue
+
+        indicator['value'] = indicator['indicator']
```
Comment (ronykoz) [inline Packs/FeedAlienVault/Integrations/FeedAlienVaultOTXTaxii/FeedAlienVaultOTXTaxii.py]: shouldn't we add type as well?

File: Packs/FeedAlienVault/Integrations/FeedAlienVaultOTXTaxii/FeedAlienVaultOTXTaxii.py [hunk 4]
```diff
@@ -0,0 +1,667 @@
...
+
+    Args:
+        sub_indicator_list(list): A list of STIXDecoded indicators
+        full_indicator_list(list): A list of all the indicators fetched to this point - used to prevent duplications.
+
+    Returns:
+        list,list. A list of parsed indicators and an updated list of all indicators polled
+    """
+    parsed_indicator_list = []  # type: List
+    for indicator in sub_indicator_list:
```
Comment (ronykoz) [inline Packs/FeedAlienVault/Integrations/FeedAlienVaultOTXTaxii/FeedAlienVaultOTXTaxii.py]: seems like you abusing the indicator here, you should create a new object which will hold the real indicator within the rawJson, the value and the type, the last two should be in the outer scope as well.
Oh and the fields too, are we sure that it's all the mapping we want to do?

File: Packs/FeedAlienVault/Integrations/FeedAlienVaultOTXTaxii/FeedAlienVaultOTXTaxii.py [hunk 5]
```diff
@@ -0,0 +1,667 @@
...
+                if limit == index:
+                    break
+
+    return indicator_list
+
+
+def main():
+    params = demisto.params()
+
+    client = Client(params.get('api_key'), params.get('collection'), params.get('insecure'), params.get('proxy'),
```
Comment (ronykoz) [inline Packs/FeedAlienVault/Integrations/FeedAlienVaultOTXTaxii/FeedAlienVaultOTXTaxii.py]: Shouldn't it be?
```suggestion
    client = Client(params.get('api_key'), params.get('collections'), params.get('insecure'), params.get('proxy'),
```

Comment (roysagi) [reply Packs/FeedAlienVault/Integrations/FeedAlienVaultOTXTaxii/FeedAlienVaultOTXTaxii.py]: done

File: Packs/FeedAlienVault/Integrations/FeedAlienVaultOTXTaxii/FeedAlienVaultOTXTaxii.py (hunk 4 above)
Comment (ronykoz) [reply Packs/FeedAlienVault/Integrations/FeedAlienVaultOTXTaxii/FeedAlienVaultOTXTaxii.py]: The value should also be part of the rawJSON

Comment (roysagi) [reply Packs/FeedAlienVault/Integrations/FeedAlienVaultOTXTaxii/FeedAlienVaultOTXTaxii.py]: done

**Comment by roysagi** [comment]
@kirbles19 Hi, If you can please check the following:
* [yml](https://github.com/demisto/content/blob/alien_vault_otx_taxii_feed/Packs/FeedAlienVault/Integrations/FeedAlienVaultOTXTaxii/FeedAlienVaultOTXTaxii.yml)
* [description](https://github.com/demisto/content/blob/alien_vault_otx_taxii_feed/Packs/FeedAlienVault/Integrations/FeedAlienVaultOTXTaxii/FeedAlienVaultOTXTaxii_description.md)

Thank you!

**Comment by kirbles19** [comment]
@roysagi Done.

**Pull request merged**